pkg/logger/logger.go

```diff
@@ -201,13 +198,11 @@ func (l *Logger) PrintReq(username, upstream string, req *http.Request, url url.
 		}
 	}
 
-	client := GetClient(req, l.reverseProxy)
-
 	l.mu.Lock()
 	defer l.mu.Unlock()
 
 	l.reqTemplate.Execute(l.writer, reqLogMessageData{
-		Client:          client,
+		Client:          req.RemoteAddr,
```

Comment:
Should this not also be using a passed in `client` rather than just the request remote address?

Reply:
That's a good question.  As it turns out, passing the real client IP through is a little painful since we don't have access to the `OAuthProxy` in the `logger` package and the `logger` doesn't get any of the `Options`.  Passing the "client" through breaks the format, as it includes the real client IP in parens.  If theres anything (ex. Logstash) relying on the format of the oauth2_proxy logs, it's this.  Especially in the light of #321 and the fact that Apache HTTPd and NGINX both (by default) do not log the real client IP, I'm not sure it makes sense to keep that behaviour.  Obviously this could be a breaking change for someone, but I'm skeptical its advantages outweigh its downsides.

I've made quite a few other changes that affect the default logging behaviour in this PR.  I was hoping to get feedback to help decide what we can change and what we can't.  Let me know what you think, I can certainly retain all the default behaviour, but I started off with what would have made the most sense to me.

Reply:
Alright, I think I've taken care of all the other feedback, but I'm still pretty concerned about this change and the other breaking change (the default is no real-client-ip header).  As I pointed out, I think what we have here makes more sense as default behaviour, but I understand that any breaking change no matter how small has to be weighed carefully.

What do you think?

Reply:
I think I need to start and read through this all again to understand the breaking changes, it's quite a large PR touching many areas which makes it harder to review. Could you link here the other breaking change?

@syscll @steakunderscore Reviews on this PR would be really useful if either of you have some time

Reply:
Just the default real client IP header being https://github.com/pusher/oauth2_proxy/pull/425/files#r393885914

Reply:
@syscll this is still my primary concern.  In order to restore the same behaviour I have to include the suggestion [here](https://github.com/pusher/oauth2_proxy/pull/425/files#r399659298) and either checkout the old code for real request logging (simpler, but duplicating some code) OR move the code for `GetClientString` and dependant functions to a new file / structure as the `loggingHandler` doesn't have access to the `OAuthProxy` object.